packages/x-date-pickers/src/internals/hooks/useField/useFieldSectionContentProps.ts

```diff
@@ -244,18 +244,23 @@ function getSectionValueText(
   switch (section.type) {
     case 'month': {
       if (section.contentType === 'digit') {
-        return adapter.format(adapter.setMonth(adapter.date(), Number(section.value) - 1), 'month');
+        const dateWithMonth = adapter.setMonth(adapter.date(), Number(section.value) - 1);
+        return adapter.isValid(dateWithMonth) ? adapter.format(dateWithMonth, 'month') : '';
       }
       const parsedDate = adapter.parse(section.value, section.format);
-      return parsedDate ? adapter.format(parsedDate, 'month') : undefined;
+      return parsedDate && adapter.isValid(parsedDate)
+        ? adapter.format(parsedDate, 'month')
+        : undefined;
```

Comment:
My bad. 🙈 
`isValid` already asserts the null case. 👍 
```suggestion
      return adapter.isValid(parsedDate) ? adapter.format(parsedDate, 'month') : undefined;
```